tests/scripts/local_monitoring_tests/rest_apis/test_list_metadata.py

```diff
@@ -0,0 +1,475 @@
+
+from helpers.fixtures import *
+from helpers.kruize import *
+from helpers.utils import *
+from helpers.import_metadata_json_validate import *
+from helpers.list_metadata_json_validate import *
+from helpers.list_metadata_json_schema import *
+from helpers.list_metadata_json_verbose_true_schema import *
+from helpers.list_metadata_json_cluster_name_without_verbose_schema import *
+from jinja2 import Environment, FileSystemLoader
+
+@pytest.mark.sanity
+def test_list_metadata_valid_datasource(cluster_type):
+    """
+    Test Description: This test validates GET /dsmetadata by passing a valid datasource name
+    """
+    input_json_file = "../json_files/import_metadata.json"
+
+    form_kruize_url(cluster_type)
+
+    response = delete_metadata(input_json_file)
+    print("delete metadata = ", response.status_code)
+
+    # Import metadata using the specified json
+    response = import_metadata(input_json_file)
+    import_metadata_json = response.json()
+
+    # Validate the json against the json schema
+    errorMsg = validate_import_metadata_json(import_metadata_json, import_metadata_json_schema)
+    assert errorMsg == ""
+
+    json_data = json.load(open(input_json_file))
+    datasource = json_data['datasource_name']
+
+    response = list_metadata(datasource)
+
+    list_metadata_json = response.json()
+    assert response.status_code == SUCCESS_200_STATUS_CODE
+
+    # Validate the json against the json schema
+    errorMsg = validate_list_metadata_json(list_metadata_json, list_metadata_json_schema)
+    assert errorMsg == ""
+
+    response = delete_metadata(input_json_file)
+    print("delete metadata = ", response.status_code)
+
+
+@pytest.mark.negative
+def test_list_metadata_without_parameters(cluster_type):
+    """
+    Test Description: This test validates GET /dsmetadata without passing any parameters
+    """
+    input_json_file = "../json_files/import_metadata.json"
+
+    form_kruize_url(cluster_type)
+
+    response = delete_metadata(input_json_file)
+    print("delete metadata = ", response.status_code)
+
+    # Import metadata using the specified json
+    response = import_metadata(input_json_file)
+    metadata_json = response.json()
+
+    # Validate the json against the json schema
+    errorMsg = validate_import_metadata_json(metadata_json, import_metadata_json_schema)
+    assert errorMsg == ""
+
+    response = list_metadata()
+
+    list_metadata_json = response.json()
+    assert response.status_code == ERROR_STATUS_CODE
+    datasource = "null"
+    assert list_metadata_json['message'] == LIST_METADATA_DATASOURCE_NAME_ERROR_MSG % datasource
+
+
+    response = delete_metadata(input_json_file)
+    print("delete metadata = ", response.status_code)
+
+
+@pytest.mark.negative
+@pytest.mark.parametrize("test_name, expected_status_code, datasource",
+                         [
+                             ("blank_datasource", 400, ""),
+                             ("null_datasource", 400, "null"),
```

Comment:
`test_type = {"blank": "", "null": "null", "invalid": "xyz"}`
used in [generate_test_data](https://github.com/kruize/autotune/blob/master/tests/scripts/helpers/utils.py#L243) for generating negative scenarios in create experiment tests

Reply:
Tried to strip the double quotes but still `null` is treated as a string - this might be because `datasource` here is an input query parameter
Also tried passing `None` but the parameter itself is getting omitted. Will further look for alternatives for `null` in Python